### PR TITLE
test(helper): enable global leak check for test cases

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

This PR enables global leak check in mocha test runner, to ensure if any global patching in test case to be cleared. Seems this is somewhat inaccurate such as some teardown in test hook is not detected, those are marked as ignored.

**Related issue (if exists):**

### CODE REVIEW

File: spec/operators/toPromise-spec.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ describe('Observable.prototype.toPromise', () => {
     Observable.of(42).toPromise().then((x: number) => {
       expect(wasCalled).to.be.true;
       expect(x).to.equal(42);
+
+      delete __root__.Rx;
```
Comment (kwonoj) [inline spec/operators/toPromise-spec.ts]: It's satisfying mocha's global leak checking, if test cade modifies global object and does not clear mocha catches it. In this test case it does not affect functionality but added for strict teardown.

**Comment by kwonoj** [comment]
It's small ergonomics changes so assume it'll be ok, but leaving it bit while to hear any other suggestion around before check this in.

**Comment by blesh** [comment]
:shipit: LGTM

**Comment by kwonoj** [comment]
Merged with 26ad3db.

**Pull request closed without merging**